File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int cntMax = 0, cnt = 0; ; 
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 1) {
                cnt++;
                cntMax = max(cntMax, cnt); 
            } else {
                cnt = 0; 
            }
        }
        return cntMax;
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int sum = 0 ;
        int maxSum = 0 ;
        for(int i = 0 ; i < nums.length ; i++){
            if(nums[i] == 0){
                maxSum = Math.max(maxSum,sum);
                sum = 0 ;
            }
            else{
                sum++ ;
            }
        }
        maxSum = Math.max(maxSum,sum);
        return maxSum ;
    }
}